src/socket.rs

```diff
@@ -269,9 +276,16 @@ where
 
     pub async fn connect_with_cid(
         &self,
-        cid: ConnectionId<P>,
+        cid_send: u16,
```

Comment:
I was noticing that different sides of the connection are used on connect and accept, and I was trying to understand the logic for why we would use one or the other. I looked at how we use it in trin, and the recv side is the natural one to use there. I'm sure you have a reason why `send` is the natural choice. I think picking one is likely to lead to confusion, either way, so I'm starting to think that the other solution Jacob suggested might be better.

The kind of mess of 100 and 101s in the test seem to align with that. It seems very easy to lose track of.

Reply:
That would have similar problems, but I can take a shot at it, and will put it in a different PR

it would probably look like this

```
func new_connect(id: u16, is_sender: bool) {
    if (is_sender) {}
    else {}
}

So I think the same confusion would happen

Reply:
Ok the alternative solution is in this PR https://github.com/ethereum/utp/pull/95